app/models/event.rb

```diff
@@ -238,6 +250,39 @@ def checkiner?(user)
     rsvps.where(user_id: user.id, checkiner: true).any?
   end
 
+  def reorder_waitlist!
+    return if historical?
+    return unless student_rsvp_limit
+
+    Rsvp.transaction do
+      unless students_at_limit?
+        number_of_open_spots = student_rsvp_limit - student_rsvps_count
+        to_be_confirmed = student_waitlist_rsvps.limit(number_of_open_spots)
+        to_be_confirmed.each do |rsvp|
+          rsvp.promote_from_waitlist!
+        end
+      end
+
+      Rsvp.transaction do
```

Comment:
I don't think the inner `Rsvp.transaction do` is needed, because we're already in a transaction